### ISSUE TITLE
Multiple different shapes with stroke & fill don't render correctly on wasm.

### ISSUE THREAD

**Issue by spectria-limina** [james-j-obrien/bevy_vector_shapes#37]
For some reason, in the below code, there are two issues on wasm32 but not on Linux:

- The stroke does not appear to be rendered at all
- The alpha blending appears to be incorrect, resulting in a more saturated shape:

![buggy_wasm](https://github.com/james-j-obrien/bevy_vector_shapes/assets/118812919/2f38f0e9-511d-425f-a0de-2cc15b3a59c5)

This is what it looks like on desktop (via XWayland on Fedora 39):

![normal_xwayland](https://github.com/james-j-obrien/bevy_vector_shapes/assets/118812919/33127ffc-972e-4e58-894d-2171b82389e4)

It seems to me, based on tweaking visibility of the strokes in the inspector, that the `hollow` shapes are not being handled correctly, and rendering as a second copy of the filled-in shapes. This is what is causing the apparently incorrect alpha blending.

This bug only reproduces when *both* the circle and square are spawned. If only one is spawned, or the shapes are made identical (two circles or two squares), the bug does not occur.

<details>
<summary>MWE Code</summary>

```rust
use bevy::{ecs::system::EntityCommands, prelude::*};
use bevy_vector_shapes::prelude::*;

const MARK_SIZE: f32 = 2.4;
const FILL_OPACITY: f32 = 0.22;
const STROKE_OPACITY: f32 = 0.75;
const STROKE_WIDTH: f32 = 0.05;

/// A placeable marker for players to reference movements during a fight.
#[derive(Copy, Clone, Debug, Hash, PartialOrd, Ord, PartialEq, Eq)]
#[derive(Component, Reflect)]
pub enum Mark {
    Circle,
    Square,
}

impl Mark {
    pub fn color(self) -> Color {
        Color::RED
    }

    pub fn is_square(self) -> bool {
        match self {
            Mark::Square => true,
            _ => false,
        }
    }
}

pub fn insert_mark<'w, 's, 'a>(
    mut entity: EntityCommands<'w, 's, 'a>,
    mark: Mark,
) -> EntityCommands<'w, 's, 'a> {
    entity.insert(SpatialBundle::default());
    entity.with_children(|parent| {
        let mut spawn_shape = |config| {
            if mark.is_square() {
                parent.spawn((ShapeBundle::rect(&config, Vec2::new(MARK_SIZE, MARK_SIZE)),));
            } else {
                parent.spawn((ShapeBundle::circle(&config, MARK_SIZE / 2.0),));
            };
        };

        spawn_shape(ShapeConfig {
            color: mark.color().with_a(STROKE_OPACITY),
            thickness: STROKE_WIDTH,
            hollow: true,
            alpha_mode: AlphaMode::Blend,
            transform: Transform::from_xyz(0.0, 0.0, -0.1),
            ..ShapeConfig::default_2d()
        });

        spawn_shape(ShapeConfig {
            color: mark.color().with_a(FILL_OPACITY),
            hollow: false,
            alpha_mode: AlphaMode::Blend,
            transform: Transform::from_xyz(0.0, 0.0, -0.2),
            ..ShapeConfig::default_2d()
        });
    });
    entity
}

fn main() -> eyre::Result<()> {
    let mut app = App::new();
    app.add_plugins(DefaultPlugins)
        .add_plugins(Shape2dPlugin::default())
        .add_systems(Startup, spawn_things);

    Ok(app.run())
}

fn spawn_things(mut commands: Commands) {
    commands.spawn(Camera2dBundle {
        projection: OrthographicProjection {
            near: -1000.0,
            far: 1000.0,
            scaling_mode: bevy::render::camera::ScalingMode::AutoMin {
                min_width: 40.0,
                min_height: 40.0,
            },
            ..default()
        },
        ..default()
    });
    insert_mark(commands.spawn_empty(), Mark::Circle);
    insert_mark(commands.spawn_empty(), Mark::Square).insert(Transform::from_xyz(0.0, 2.0, 0.0));
}
```
</details>

**Comment by james-j-obrien** [comment]
Very strange, I only have minimal special cases for wasm and I'm not sure how they would produce this. Will take a closer look this evening, thanks for the report!

**Comment by james-j-obrien** [comment]
Can definitely reproduce the issue but no idea why it's happening, must be some strange quirk of the `webgl2` backend as I'm not doing any special casing that would affect it. Tested on bevy 0.13 and that also doesn't resolve it although I would have been surprised if it did.

The problem is probably related to the shaders in some way, will try a few things and see if anything changes.

**Comment by james-j-obrien** [comment]
Nevermind, it was a really simple typo in the `ShapePipelineKey`, will be releasing a patch that fixes it shortly.

**Comment by james-j-obrien** [comment]
I first saw it working on bevy 0.13 and wasn't able to port it back to main so there must have been some underlying change in bevy or wgpu that is related to this. At the very least it will be fixed in the release for 0.13:

![image](https://github.com/james-j-obrien/bevy_vector_shapes/assets/30707409/88c3d489-2393-4ebc-904b-ba7b1a7fd85b)

**Comment by james-j-obrien** [comment]
Fixed in 0.7.0
